presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -92,7 +92,7 @@ public void run()
         boolean hasQuery = !Strings.isNullOrEmpty(clientOptions.execute);
         boolean isFromFile = !Strings.isNullOrEmpty(clientOptions.file);
 
-        if (!hasQuery || !isFromFile) {
+        if (!hasQuery && !isFromFile) {
```

Comment:
So, ignoring Windows, which we don't support and might not work anyway, I'm tempted to just remove it entirely, along with the detection code in `Console.detectRealTerminal()` which as taken from Jansi. But for now, we'll go with this simpler fix.